src/settings.cpp

```diff
@@ -2026,12 +2021,12 @@ uint GetCompanySettingIndex(const char *name)
  * @param force_newgame force the newgame settings
  * @note Strings WILL NOT be synced over the network
  */
-bool SetSettingValue(const StringSettingDesc *sd, const char *value, bool force_newgame)
+bool SetSettingValue(const StringSettingDesc *sd, std::string value, bool force_newgame)
 {
 	assert(sd->save.conv & SLF_NO_NETWORK_SYNC);
 
-	if (GetVarMemType(sd->save.conv) == SLE_VAR_STRQ && strcmp(value, "(null)") == 0) {
-		value = nullptr;
+	if (GetVarMemType(sd->save.conv) == SLE_VAR_STRQ && value.compare("(null)") == 0) {
```

Comment:
just curious: `value == "(null)"` doesn't work for std::string?

Reply:
It would work. Though I have noticed in myself that I make quite a few errors changing these from `compare` to `==`, so I rather do the safe thing in changing this code.

Reply:
what do you mean with "quite a few errors"? What is the problem that occurs?